### PR TITLE
Kyverno Verify Attestation - Cue template

### PR DESCRIPTION
**Pull request by pxp928**
Attestation verification to ensure that the provenance file is signed is working. Jim had to use a custom image to get that fixed. Working with him to check on the why the key-value pair check is not working as intended. I will create a patch PR to address that in the future. Currently that is not a blocker.

### CODE REVIEW

File: platform/30-kyverno-setup.sh [hunk 1]
```diff
@@ -55,5 +57,9 @@ kubectl patch \
   --type json --patch-file "${GIT_ROOT}"/platform/components/kyverno/patch_container_args.json
 
 echo -e "${C_GREEN}Creating verify-image admission control policy...${C_RESET_ALL}"
-cue export "$KYVERNO_RESOURCE_DIR"/admission-control-verify-image-resources.cue -e template -t key="$USERPUBKEY" --out yaml > "$KYVERNO_RESOURCE_DIR"/admission-control-verify-image-resources.yaml
-kubectl apply -f "$KYVERNO_RESOURCE_DIR"
+pushd "$KYVERNO_RESOURCE_DIR"
+cue export -e 'ImageClusterPolicy["verify-image"]' -t repo="$REPO" -t key="$USERPUBKEY" > $KYVERNO_RESOURCE_DIR/released/verify-signature.json
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:61]:
```suggestion
cue export -e 'ImageClusterPolicy["verify-image"]' -t repo="$REPO" -t key="$USERPUBKEY" > "$KYVERNO_RESOURCE_DIR"/released/verify-signature.json
```

File: platform/30-kyverno-setup.sh [hunk 2]
```diff
@@ -55,5 +57,9 @@ kubectl patch \
...
+cue export -e 'AttestationClusterPolicy["attest-code-review"]' -t repo="$REPO" -t key="$USERPUBKEY" > $KYVERNO_RESOURCE_DIR/released/verify-attestation.json
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:62]:
```suggestion
cue export -e 'AttestationClusterPolicy["attest-code-review"]' -t repo="$REPO" -t key="$USERPUBKEY" > "$KYVERNO_RESOURCE_DIR"/released/verify-attestation.json
```

File: platform/30-kyverno-setup.sh [hunk 3]
```diff
@@ -55,5 +57,9 @@ kubectl patch \
...
+cue export -e 'configMap["keys"]' -t repo="$REPO" -t key="$USERPUBKEY" > $KYVERNO_RESOURCE_DIR/released/configmap-keys.json
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:63]:
```suggestion
cue export -e 'configMap["keys"]' -t repo="$REPO" -t key="$USERPUBKEY" > "$KYVERNO_RESOURCE_DIR"/released/configmap-keys.json
```

File: platform/30-kyverno-setup.sh [hunk 4]
```diff
@@ -55,5 +57,9 @@ kubectl patch \
...
+popd
+kubectl apply -f $KYVERNO_RESOURCE_DIR/released
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:65]:
```suggestion
kubectl apply -f "$KYVERNO_RESOURCE_DIR"/released
```

**Review by bradbeck — APPROVED** [review]
The setup script leaves `configmap-keys.json` in a modified state.

Otherwise it seems to function as expected.

**Comment by pxp928** [comment]
> 

Ah yes I was testing with my key and forgot to re-run with the original repo key. Fixed now. Good catch.

File: resources/kyverno/admission-control-policy/kube.cue [hunk 5]
```diff
@@ -0,0 +1,99 @@
+package kube
```
Comment (bradbeck) [inline resources/kyverno/admission-control-policy/kube.cue:1]: I wonder if the package should be something like "kyverno" or "kyvernoAdmissionControlPolicy"?

Comment (pxp928) [reply resources/kyverno/admission-control-policy/kube.cue:1]: I agree. I made it too generic I will change that. I think `kyverno` should suffice.

File: resources/kyverno/admission-control-policy/kube.cue [hunk 6]
```diff
@@ -0,0 +1,99 @@
+package kyverno
```
Comment (bradbeck) [inline resources/kyverno/admission-control-policy/kube.cue:1]: Maybe name this file `kyverno.cue` now?

**Review by bradbeck — APPROVED** [review]
/lgtm

**Review by mlieberman85 — APPROVED** [review]

**Pull request merged**